devtools/zenodo/zenodo_data_release.py

```diff
@@ -84,13 +86,31 @@ def __init__(self, env: str):
 
         logger.info(f"Using Zenodo token: {token[:4]}...{token[-4:]}")
 
+    def retry_request(self, *, method, url, max_tries=5, **kwargs):
+        """Wrap requests.request in retry logic.
+
+        Passes method, url, and **kwargs to requests.request.
+        """
+        base_timeout = 2
+        for try_num in range(1, max_tries):
+            try:
+                return requests.request(method=method, url=url, **kwargs)
+            except requests.RequestException as e:
+                timeout = base_timeout**try_num
+                logger.warning(f"Attempt #{try_num} Got {e}, retrying in {timeout} s")
+                time.sleep(timeout)
+
+        # don't catch errors on the last try.
+        return requests.request(method=method, url=url, **kwargs)
+
     def get_deposition(self, deposition_id: int) -> _LegacyDeposition:
         """LEGACY API: Get JSON describing a deposition.
 
         Depositions can be published *or* unpublished.
         """
-        response = requests.get(
-            f"{self.base_url}/deposit/depositions/{deposition_id}",
+        response = self.retry_request(
+            method="GET",
+            url=f"{self.base_url}/deposit/depositions/{deposition_id}",
             headers=self.auth_headers,
             timeout=5,
```

Comment:
I don't think it really matters, but it seems funny that we start with a timeout of 5 seconds, and then we have to retry, we do with `2**1` (2) seconds, and then `2**2`(4) seconds.  Or does the `timeout` that's hiding inside of `**kwargs` not get overwritten when `timeout` is assigned in the `except` clause?